### DISCUSSION TITLE
Twitter oAuth

### DISCUSSION THREAD

**Discussion by omergr8** [supabase/supabase#2270] [Questions]
Hi, Twitter oAuth is currently not working. Anyone facing issue with twitter. Every other third party oAuth is working except for twitter.
I get following error:
![IMG-20210708-WA0001](https://user-images.githubusercontent.com/62819343/124833384-8d620880-df97-11eb-9546-847912e14432.jpg)

**Comment by gtokman** [comment]
Can you share more info on how you are trying to authenticate with Twitter? Code samples or steps to reproduce?

**Reply by omergr8** [reply]
I'm using following method:
const { user, session, error } = await supabase.auth.signIn({
  provider: 'twitter'
})
I'm getting error only for twitter, if I put anything else like google, discord, Facebook, it's working fine.

**Reply by gtokman** [reply]
I can also reproduce this error too.

I checked Twitter and 3 leg auth is enabled and configured.

But as the author said, calling `signIn({provider: 'twitter'})` returns this error.
```
{
message: "no Route matched with those values"
}
```

**Reply by gtokman** [reply]
Actually, I was able to fix the issue. Make sure you go into auth and enable Twitter login. Enter these keys into Supabase.
<img width="638" alt="Screen Shot 2021-07-11 at 2 41 47 PM" src="https://user-images.githubusercontent.com/12258850/125206640-46ee0180-e256-11eb-9912-514db48c1706.png">

**Reply by inian** [reply]
Here is a video guide on setting up Twitter Auth - https://www.youtube.com/watch?v=z1rWsRb4PDw

Let me know if there are still issues in setting this up. Marking as answered since @gtokman was able to solve their issue

**Reply by MisterJimson** [reply]
I'm also struggling to get this working. Twitter has changed the developer portal and the steps listed in the video no longer work.

**Reply by kevcodez** [reply]
Running into the same issue @MisterJimson did you find a solution?

**Reply by kevcodez** [reply]
Solved it using these settings

![image](https://user-images.githubusercontent.com/14073399/148678876-9a4fcb3a-b659-4036-a6b3-82bff731a287.png)

And using the API key and secret, rather than the client id/secret

**Reply by MisterJimson** [reply]
Thanks! I'll give it a try.

**Reply by burggraf** [reply]
No matter what I do now, I now still get `{"message":"no Route matched with those values"}`.  I wrote the original documentation for this, but Twitter has completely changed this.  I've tried the settings shown above, but it doesn't work for me, whether I use `OAuth 2.0 Client ID and Client Secret` or `Access Token and Secret`.  What's going on with Twitter?

**Reply by Timonzimm** [reply]
In my case the problem was indeed tta I was using the `OAuth 2.0 Client ID and Client Secret` and not the `Access Token and Secret`.

But anyway it's a bit annoying because you cannot then make requests to Twitter API on the behalf of the user, issue discussed here: https://github.com/supabase/supabase/discussions/2961

**Reply by stelateral** [reply]
Same here, using `API Key and Secret` rather than` OAuth 2.0 Client ID and Client Secret` worked for Twitter Authentication with Supabase.

**Reply by philmade** [reply]
I managed to get this working also by using the twitter API key and secret. What's strange is that it apparently grants us access to read tweets as well as profile information. If we just want login via twitter, why do we also have to have access to user tweets also? This seems strange.

**Reply by petrbela** [reply]
Twitter v1 oauth (using api key/secret) doesn't have usual scopes like v2 oauth, instead, it has just three permission levels (read-only, read-write, and read-write-dms) that will be the same for every token.

Twitter has added the standard v2 oauth (client id/secret) a few years ago but the rollout has been very slow because v1 was, and still is, widely used. Moreover, some API endpoints (such as posting with attachments) are not available yet in the v2 api, so most providers still use Twitter's v1 auth.
